src/service/video-manager-impl.js

```diff
@@ -185,11 +184,32 @@ export class VideoManager {
     if (!video.supportsPlatform()) {
       return;
     }
+    if (!this.viewportObserver_) {
+      const viewportCallback = (
+        /** @type {!Array<!IntersectionObserverEntry>} */ records
+      ) =>
+        records.forEach(({target, isIntersecting}) => {
+          this.getEntry_(target).updateVisibility(
+            /* isVisible */ isIntersecting
+          );
+        });
+      this.viewportObserver_ = getViewportObserver(
+        viewportCallback,
+        this.ampdoc.win
+      );
+    }
+    this.viewportObserver_.observe(
```

Comment:
We'd need the video elements to call the video manager on their layout/unlayout calls. The best way might unfortunatley to monkey patch their unlayout/layout cbs within the manager. I'll try it now

Reply:
Hmm. Why not just call directly from `unlayoutCallback`?

Reply:
that works too! small change to the 11 video components

Reply:
Yeah. Let's defer actually putting it in the `unlayout` callback since that might have some unintended consequences. As soon as we pair build with something, we'll follow up.